### PR TITLE
Merge Develop into master

### PR DESCRIPTION
**Pull request by lakchote**

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -23,7 +23,15 @@ task("accounts", "Prints the list of accounts", async (taskArgs, hre) => {
 // Go to https://hardhat.org/config/ to learn more
 
 const config: HardhatUserConfig = {
-  solidity: "0.8.4",
+  solidity: {
+    version: "0.8.4",
+    settings: {
+      optimizer: {
+        enabled: true,
+        runs: 10,
```
Comment (vidvidvid) [inline hardhat.config.ts:31]: would you care to explain how did you pick the value of `runs` param?

File: tests/Ballot/Ballot.ts [hunk 2]
```diff
@@ -86,72 +110,115 @@ describe("Ballot", function () {
   });
 
   describe("when the voter interact with the vote function in the contract", function () {
-    // TODO
-    it("is not implemented", async function () {
-      throw new Error("Not implemented");
+    it("should has the right to vote", async function () {
```
Comment (vidvidvid) [inline tests/Ballot/Ballot.ts:113]: I think what you meant was:
```suggestion
    it("should not have the right to vote", async function () {
```

**Review by vidvidvid — COMMENTED** [review]
some minor suggestions/questions, otherwise it looks good 👍

File: hardhat.config.ts (hunk 1 above)
Comment (lakchote) [reply hardhat.config.ts:31]: It depends if your contract functions are going to get called a lot of times or not.
The higher the runs number, the heavier the cost of deployment will be but the contract function calls with cost less gas.
Inversely the lower the runs number, cost of deployment will be less but function calls will cost a higher amount of gas.

So it's a tradeoff to be made.
Regarding the ballot, it seems to be a good target.

File: tests/Ballot/Ballot.ts (hunk 2 above)
Comment (lakchote) [reply tests/Ballot/Ballot.ts]: It's true it was a typo.

File: hardhat.config.ts (hunk 1 above)
Comment (vidvidvid) [reply hardhat.config.ts:31]: thank you for the explanation - still, i gotta look more into some real examples of how these numbers are set.

**Pull request merged**